src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,92 @@
-export const PeopleFilters = () => {
+import { Link, useSearchParams } from 'react-router-dom';
+import { Person } from '../types/Person';
+import React, { useState, useEffect } from 'react';
+
+interface PeopleFiltersProps {
+  persons: Person[];
+  onFilterChange: (filter: 'all' | 'male' | 'female') => void;
+  onCenturyChange: (century: string | 'all' | null) => void;
+  onSearchChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+export const PeopleFilters: React.FC<PeopleFiltersProps> = ({
+  onFilterChange,
+  onCenturyChange,
+  onSearchChange,
+}) => {
+  const [filter, setFilter] = useState<'all' | 'male' | 'female'>('all');
+  const [searchQuery, setSearchQuery] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
+  const centuryQuery = searchParams.getAll('century') || [];
+
+  useEffect(() => {
+    const currentFilter = searchParams.get('filter') as
+      | 'all'
+      | 'male'
+      | 'female';
+    const currentSearch = searchParams.get('search') || '';
+
+    if (currentFilter) {
+      setFilter(currentFilter);
+    }
+
+    setSearchQuery(currentSearch);
+  }, [searchParams, centuryQuery]);
```

Comment:
Including `centuryQuery` in the dependency array of `useEffect` might cause unnecessary re-renders if `centuryQuery` changes. If `centuryQuery` is not directly used in the effect, consider removing it from the dependency array.